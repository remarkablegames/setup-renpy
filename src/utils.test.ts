import os from 'os';

import {
  getAddonDirectory,
  getBinaryDirectory,
  getBinaryPath,
  getDownloadObject,
  getLauncherDirectory,
} from './utils';

jest.mock('os');
const mockedOs = jest.mocked(os);

const platforms = ['darwin', 'linux', 'win32'] as const;
const architectures = ['arm', 'x32', 'x64'] as const;

const table = platforms.reduce(
  (testSuites, os) => [
    ...testSuites,
    ...architectures.map((arch) => [os, arch] as [string, string]),
  ],
  [] as [string, string][],
);

const version = '8.2.1';

describe('getDownloadObject', () => {
  describe.each(table)('when OS is %p and arch is %p', (os, arch) => {
    beforeEach(() => {
      jest.resetAllMocks();
      mockedOs.platform.mockReturnValueOnce(os as NodeJS.Platform);
      mockedOs.arch.mockReturnValueOnce(arch);
    });

    it('gets download object', () => {
      expect(getDownloadObject(version)).toMatchSnapshot();
    });
  });
});

describe('getBinaryPath', () => {
  describe.each(platforms)('when OS is %p', (os) => {
    beforeEach(() => {
      jest.resetAllMocks();
      mockedOs.platform.mockReturnValueOnce(os);
    });

    it('returns CLI filepath', () => {
      const directory = 'directory';
      const name = 'name';
      expect(getBinaryPath(directory, name)).toMatchSnapshot();
    });
  });
});

describe('getBinaryDirectory', () => {
  it.each(architectures)('returns CLI directory for arch %p', (arch) => {
    mockedOs.arch.mockReturnValueOnce(arch);
    const directory = 'directory';
    expect(getBinaryDirectory(directory, version)).toMatchSnapshot();
  });
});

describe('getLauncherDirectory', () => {
  it('returns launcher directory', () => {
    const directory = 'directory';
    expect(getLauncherDirectory(directory)).toMatchSnapshot();
  });
});

describe('getAddonDirectory', () => {
  it.each(['rapt', 'renios', 'web'])(
    'returns addon directory for %p',
    (addon) => {
      const directory = 'directory';
      expect(getAddonDirectory(directory, addon)).toMatchSnapshot();
    },
  );
});
