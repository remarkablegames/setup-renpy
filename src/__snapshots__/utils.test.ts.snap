// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`createLauncherBinary creates launcher binary on "darwin" 1`] = `
[
  [
    "directory/name",
    "cliPath directory/launcher "$@"",
  ],
]
`;

exports[`createLauncherBinary creates launcher binary on "linux" 1`] = `
[
  [
    "directory/name",
    "cliPath directory/launcher "$@"",
  ],
]
`;

exports[`createLauncherBinary creates launcher binary on win32 and version "7.8.5" 1`] = `
[
  [
    "directory/name.bat",
    "directory/lib/py2-windows-x86_64/python.exe directory/renpy.py directory/launcher %*",
  ],
]
`;

exports[`createLauncherBinary creates launcher binary on win32 and version "8.2.3" 1`] = `
[
  [
    "directory/name.bat",
    "directory/lib/py3-windows-x86_64/python.exe directory/renpy.py directory/launcher %*",
  ],
]
`;

exports[`getBinaryDirectory returns CLI directory for arch "arm" 1`] = `"directory/renpy-8.2.1-sdkarm"`;

exports[`getBinaryDirectory returns CLI directory for arch "x64" 1`] = `"directory/renpy-8.2.1-sdk"`;

exports[`getBinaryPath returns CLI path 1`] = `"directory/name.sh"`;

exports[`getDownloadObject when arch is "arm" gets download object 1`] = `
{
  "rapt": "https://www.renpy.org/dl/8.2.1/renpy-8.2.1-rapt.zip",
  "renios": "https://www.renpy.org/dl/8.2.1/renpy-8.2.1-renios.zip",
  "sdk": "https://www.renpy.org/dl/8.2.1/renpy-8.2.1-sdkarm.tar.bz2",
  "web": "https://www.renpy.org/dl/8.2.1/renpy-8.2.1-web.zip",
}
`;

exports[`getDownloadObject when arch is "x64" gets download object 1`] = `
{
  "rapt": "https://www.renpy.org/dl/8.2.1/renpy-8.2.1-rapt.zip",
  "renios": "https://www.renpy.org/dl/8.2.1/renpy-8.2.1-renios.zip",
  "sdk": "https://www.renpy.org/dl/8.2.1/renpy-8.2.1-sdk.zip",
  "web": "https://www.renpy.org/dl/8.2.1/renpy-8.2.1-web.zip",
}
`;

exports[`getLauncherDirectory returns launcher directory 1`] = `"directory/launcher"`;
